factory/usr/lib/the-modeenv

```diff
@@ -5,10 +5,19 @@ cp /sysroot/run/image.fstab /run/
 # TODO:UC20: move this logic into snap-bootstrap to be incorporated with 
 # initramfs-mounts
 
+if grep -q snapd_recovery_mode=run /proc/cmdline; then
+    run_mode=1
+elif grep -q snapd_recovery_mode=install /proc/cmdline; then
+    if grep -q mode=run /run/mnt/ubuntu-data/system-data/var/lib/snapd/modeenv; then
```

Comment:
this condition is too restrictive.
What if `snapd_recovery_mode=factory-reset`?